### PR TITLE
feat: support for Azure quota rule

### PR DESCRIPTION
**Pull request by mattwelke**
## Issue
Closes https://github.com/validator-labs/validatorctl/issues/225.

## Description
Adds support for the rule. Like other rules with potentially repetitive input, users can provide input with a JSON file too.

## Extras

* Fixes a bug where if a user said they didn't want to enter Azure community image gallery rules, it would force them to anyways.
* Fixes a bug where when users were prompted for a permission set for an Azure RBAC rule, the prompt incorrectly called it an IAM (for AWS) rule.

### CODE REVIEW

**Review by TylerGillson — COMMENTED** [review]
Since you have fresh context, can you please take a look at https://github.com/validator-labs/validatorctl/issues/29 and either address it as part of this PR or as a follow up?

File: pkg/services/validator/azure.go [hunk 1]
```diff
@@ -416,6 +424,130 @@ func readCommunityGalleryImageRule(c *components.AzurePluginConfig, r *plug.Comm
...
+		c.Validator.QuotaRules[idx] = *r
+	}
+	return nil
+}
+
+// readQuotaRuleResourceSets begins the process of setting the resource sets of the rule. Users can
+// provide input via file or prompts.
+// nolint:dupl
+func readQuotaRuleResourceSets(r *plug.QuotaRule) error {
+	log.InfoCLI("Note: You must configure at least one resource set for rule.")
```
Comment (TylerGillson) [inline pkg/services/validator/azure.go:503]: per rule?

File: pkg/services/validator/azure.go [hunk 2]
```diff
@@ -416,6 +424,130 @@ func readCommunityGalleryImageRule(c *components.AzurePluginConfig, r *plug.Comm
...
+			time.Sleep(2 * time.Second)
+			resourceSetFile, err := prompts.EditFileValidatedByFullContent(cfg.AzurePermissionSetPrompt, "", prompts.ValidateJSON, 1)
+			if err != nil {
+				return fmt.Errorf("failed to configure resource sets: %w", err)
+			}
+			resourceSetBytes = []byte(resourceSetFile)
+		}
+
+		var resourceSets []plug.ResourceSet
+		errUnmarshal := json.Unmarshal(resourceSetBytes, &resourceSets)
```
Comment (TylerGillson) [inline pkg/services/validator/azure.go:533]: nit: could you unmarshal directly into `r.ResourceSets`?

**Comment by mattwelke** [comment]
> Since you have fresh context, can you please take a look at #29 and either address it as part of this PR or as a follow up?

Sure, I'll work on that next and add it to this PR as an extra bug fix.

**Review by TylerGillson — APPROVED** [review]

**Pull request merged**